config/config.go

```diff
@@ -62,33 +63,49 @@ func FindAndReadConfig(fs afero.Fs, configFile string) (*v2.Config, error) {
 	if err != nil {
 		return nil, err
 	}
+
 	switch v {
-	case 1:
+	case 1: //Upgrade the config version
+
 		c, err := v1.ReadConfig(b)
 		if err != nil {
 			return nil, err
 		}
 
-		c2, err := UpgradeConfigVersion(c)
-		return c2, err
-
+		return UpgradeConfigVersion(c)
 	case 2:
 
-		c2, err := v2.ReadConfig(b)
-		return c2, err
+		return v2.ReadConfig(b, fs, configFile)
 	default:
 		return nil, errs.NewUser("could not figure out config file version")
 	}
+
 }
 
 type ver struct {
-	Version int `json:"version"`
+	Version int `json:"version" yaml:"version"`
 }
 
 // detectVersion will detect the version of a config
-func detectVersion(b []byte) (int, error) {
+func detectVersion(b []byte, fs afero.Fs, configFile string) (int, error) {
 	v := &ver{}
-	err := json.Unmarshal(b, v)
+	var err error
+
+	info, err := fs.Stat(configFile)
+	if err != nil {
+		return 0, errs.WrapUser(err, "unable to find file")
+	}
+
+	//Unmarshals based on file extension
+	switch filepath.Ext(info.Name()) {
+	case ".yml":
```

Comment:
```suggestion
	case ".yml", ".yaml":
```
Both are [common](https://yaml.org/faq.html) extensions